### PR TITLE
[FLINK-7185] Activate checkstyle flink-java/io

### PR DESCRIPTION
**Pull request by dawidwys**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java [hunk 1]
```diff
@@ -21,13 +21,18 @@
...
 import org.apache.flink.types.parser.FieldParser;
 import org.apache.flink.util.Preconditions;
+import org.apache.flink.util.StringUtils;
 
 import java.io.IOException;
-import org.apache.flink.core.fs.Path;
-import org.apache.flink.util.StringUtils;
 
+/**
+ * InputFormat thar reads csv files.
```
Comment (zentol) [inline flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java]: thar -> that

File: flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java [hunk 2]
```diff
@@ -23,9 +23,32 @@
...
+import org.apache.flink.api.java.tuple.Tuple10;
+import org.apache.flink.api.java.tuple.Tuple11;
+import org.apache.flink.api.java.tuple.Tuple12;
+import org.apache.flink.api.java.tuple.Tuple13;
+import org.apache.flink.api.java.tuple.Tuple14;
+import org.apache.flink.api.java.tuple.Tuple15;
+import org.apache.flink.api.java.tuple.Tuple16;
+import org.apache.flink.api.java.tuple.Tuple17;
+import org.apache.flink.api.java.tuple.Tuple18;
+import org.apache.flink.api.java.tuple.Tuple19;
```
Comment (zentol) [inline flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java]: ah...now i see why the star import is useful here; the tuple generator doesn't have to modify the import statements.

File: flink-java/src/main/java/org/apache/flink/api/java/io/PrintingOutputFormat.java [hunk 3]
```diff
@@ -18,12 +18,16 @@
...
-import java.io.PrintStream;
-
 import org.apache.flink.annotation.PublicEvolving;
 import org.apache.flink.api.common.io.RichOutputFormat;
 import org.apache.flink.configuration.Configuration;
 
+import java.io.PrintStream;
+
+/**
+ * Output format that prints results into underlying {@link PrintStream}.
```
Comment (zentol) [inline flink-java/src/main/java/org/apache/flink/api/java/io/PrintingOutputFormat.java]: I would reword this to mention stdout/stderr; the usage of `PrintStream` is an (unimportant) implementation detail.

File: flink-java/src/main/java/org/apache/flink/api/java/io/TextOutputFormat.java [hunk 4]
```diff
@@ -18,65 +18,73 @@
...
 import java.nio.charset.Charset;
 import java.nio.charset.IllegalCharsetNameException;
 import java.nio.charset.UnsupportedCharsetException;
 
-import org.apache.flink.annotation.PublicEvolving;
-import org.apache.flink.api.common.io.FileOutputFormat;
-import org.apache.flink.core.fs.Path;
-
+/**
+ * {@link FileOutputFormat} that stores values by calling {@link Object#toString()} method.
```
Comment (zentol) [inline flink-java/src/main/java/org/apache/flink/api/java/io/TextOutputFormat.java]: This isn't necessarily correct, as a `TextFormatter` might be used.

How about
```
A {@link FileOutputFormat} that writes objects to a text file.

<p>Objects are converted to Strings using either {@link Object#toString()} or a {@link TextFormatter}.
```

File: flink-java/src/main/java/org/apache/flink/api/java/io/TextOutputFormat.java [hunk 5]
```diff
@@ -18,65 +18,73 @@
...
+
 	private transient Charset charset;
 
 	// --------------------------------------------------------------------------------------------
 
-	public static interface TextFormatter<IN> extends Serializable {
-		public String format(IN value);
+
+	/**
+	 * Formatter that transforms values into its {@link String} representations.
```
Comment (zentol) [inline flink-java/src/main/java/org/apache/flink/api/java/io/TextOutputFormat.java]: into its -> into their

File: flink-java/src/test/java/org/apache/flink/api/java/io/CollectionInputFormatTest.java [hunk 6]
```diff
@@ -90,18 +94,18 @@ public String toString() {
 	@Test
 	public void testSerializability() {
 		try (ByteArrayOutputStream buffer = new ByteArrayOutputStream();
-			 ObjectOutputStream out = new ObjectOutputStream(buffer)) {
+				ObjectOutputStream out = new ObjectOutputStream(buffer)) {
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/CollectionInputFormatTest.java]: indentation seems off.

File: flink-java/src/test/java/org/apache/flink/api/java/io/CsvInputFormatTest.java [hunk 7]
```diff
@@ -224,34 +230,31 @@ private void ignoreInvalidLines(int bufferSize) {
...
 	@Test
 	public void ignoreSingleCharPrefixComments() {
 		try {
 			final String fileContent = "#description of the data\n" +
-									   "#successive commented line\n" +
-									   "this is|1|2.0|\n" +
-									   "a test|3|4.0|\n" +
-									   "#next|5|6.0|\n";
-			
+										"#successive commented line\n" + "this is|1|2.0|\n" + "a test|3|4.0|\n" + "#next|5|6.0|\n";
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/CsvInputFormatTest.java]: lets move this line more to the left and keep them in separate lines

File: flink-java/src/test/java/org/apache/flink/api/java/io/CsvInputFormatTest.java [hunk 8]
```diff
@@ -266,42 +269,41 @@ public void ignoreSingleCharPrefixComments() {
...
-			
-			
+
 			final String fileContent = "//description of the data\n" +
-									   "//successive commented line\n" +
-									   "this is|1|2.0|\n"+
-									   "a test|3|4.0|\n" +
-									   "//next|5|6.0|\n";
-			
+										"//successive commented line\n" +
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/CsvInputFormatTest.java]: move more to the left

File: flink-java/src/test/java/org/apache/flink/api/java/io/FromElementsTest.java [hunk 9]
```diff
@@ -18,8 +18,12 @@
 package org.apache.flink.api.java.io;
 
 import org.apache.flink.api.java.ExecutionEnvironment;
+
 import org.junit.Test;
 
+/**
+ * Tests for {@link ExecutionEnvironment::fromElements}.
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/FromElementsTest.java]: is this a valid javadoc notation?

File: flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java (hunk 2 above)
Comment (dawidwys) [inline flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java]: Oh, ok. I didn't notice, it was because of code generation previously. How do we want to proceed with it then? Do we want to add suppression for it? Just adding the `//CHECKSTYLE: OFF` won't work as it violates the `ImportOrder` rule.

File: flink-java/src/test/java/org/apache/flink/api/java/io/FromElementsTest.java [hunk 10]
```diff
@@ -22,7 +22,7 @@
 import org.junit.Test;
 
 /**
- * Tests for {@link ExecutionEnvironment::fromElements}.
+ * Tests for {@link ExecutionEnvironment#fromElements}}.
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/FromElementsTest.java]: duplicate `}`

File: flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java (hunk 2 above)
Comment (zentol) [inline flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java]: We will have to disable them in place as we otherwise disable entire rules for some files. You can specify multiple rules to disable like this: `//CHECKSTYLE.OFF: AvoidStarImport|ImportOrder`. At least that should work.

Comment (dawidwys) [inline flink-java/src/main/java/org/apache/flink/api/java/io/CsvReader.java]: Yes, previously I tried:

    //CHECKSTYLE OFF: AvoidStarImport
    import org.apache.flink.api.java.tuple.*;
    //CHECKSTYLE ON: AvoidStarImport

and it did not work as the comment line was treated as an empty line between import blocks.

It worked though with the comments at the end of the line. Sorry I did not try it earlier.

**Comment by zentol** [comment]
merging.
